src/watcher.js

```diff
@@ -0,0 +1,38 @@
+/* @flow */
+import Watchpack from 'watchpack';
+import debounce from 'debounce';
+
+import {createLogger} from './util/logger';
+
+const log = createLogger(__filename);
+
+
+export default function onSourceChange(
+    {sourceDir, artifactsDir, onChange}: Object): Watchpack {
+  // TODO: For network disks, we would need to add {poll: true}.
+  const watcher = new Watchpack();
+
+  const executeImmediately = true;
+  watcher.on('change', debounce((filePath) => {
```

Comment:
Nit: how about extracting the debounce callback for readability? (probably for the same reason, `executeImmediately` is defined at line 15 instead of being passed as a `true` literal value in the `debounce` parameters), e.g. something like:

```js
  const executeImmediately = true;
  const onFileChange = (filePath) => {
    proxyFileChanges({artifactsDir, onChange, filePath});
  };
  watcher.on('change', debounce(onFileChange, 1000, executeImmediately));
```